contracts/src/spaces/entitlements/poap/PoapEntitlement.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+// interfaces
+
+// libraries
+
+// contracts
+
+contract PoapEntitlement {
+  IPOAP immutable poapContract;
+
+  constructor(address poapContractAddress) {
+    poapContract = IPOAP(poapContractAddress);
+  }
+
+  function isEntitled(
+    address[] calldata users,
+    bytes calldata data
+  ) external view returns (bool) {
+    uint256 eventId = abi.decode(data, (uint256));
+
+    for (uint256 i = 0; i < users.length; i++) {
+      if (_hasEventPoap(users[i], eventId)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  function schema() public pure returns (string memory) {
+    return "uint256 eventId";
```

Comment:
For a parameter list with multiple params, would this be comma-separated?
e.g.

"uint256 eventId, address someContractAddress"

Reply:
right, here's an example: https://sepolia.easscan.org/schema/view/0x4f532d4a635375689656f9fe828c8baaa7d475c90e2dbcd381580d65598244f8

Reply:
Is this an EIP based way of doing this? It's basically just using method selector encoding while retaining the variable name. If it's a standard, then I guess we can start with this, though I'd like to also have a description. Let's hop on a call.